faq/sql-faq.md

```diff
@@ -7,6 +7,29 @@ summary: Learn about the FAQs related to TiDB SQL.
 
 This document summarizes the FAQs related to SQL operations in TiDB.
 
+## Does TiDB support secondary key?
+
+Yes. You can have the [`NOT NULL` constraint](/constraints.md#not-null) on a non-primary key column with a unique [secondary index](/develop/dev-guide-create-secondary-indexes.md). In this case, the column works as a secondary key.
+
+## How does TiDB perform when executing DDL operations on a large table?
+
+DDL operations of TiDB on large tables are usually not an issue. TiDB supports online DDL operations, and these DDL operations do not block DML operations.
+
+For some DDL operations such as adding columns, deleting columns or dropping indexes, TiDB can perform these operations quickly.
+
+For some heavy DDL operations such as adding indexes, TiDB needs to backfill data, which takes a longer time (depending on the size of the table) and consumes additional resources. The impact on online traffic is tunable. TiDB can do the backfill with multiple threads, and the number of threads is set by the following system variables:
```

Comment:
`the number of threads is set by the following system variables` -> `the resource consumed can be set by the following system variables`